File: Java/Language/Basics/src/CurrentAccount.java
package banking;

//non-extendible class
final class CurrentAccount extends Account {

	public void deposit(double amount) {
		if(balance < 0)
			amount *= 0.9;
		balance += amount;
	}
	
	public void withdraw(double amount) throws InsufficientFundsException {
		balance -= amount;
	}
}

File: CSharp/Basics/AbstractClassTest/Banking/CurrentAccount.cs
namespace Banking
{
	sealed class CurrentAccount : Account
	{
		public override void Deposit(decimal amount)
		{
			if(Balance < 0)
				amount *= 0.9M;
			Balance += amount;
		}
		
		public override void Withdraw(decimal amount)
		{
			Balance -= amount;
		}
	}
}

